File: pylabs/lab1.py

from math import *

a = float(input("Введите a: "))
x = float(input("Введите x: "))
fun = str(input("Введите ф-ию которую вы хотите вычислить( g, f, y): "))
if (fun == 'g'):
    g = (-1 * (2 * (-5 * a**2 + 3 * a * x + 2 * x**2) / (5 * a**2 + 9 * a * x - 2 * x**2)))
    print(g)
elif (fun == 'f'):
    f = (sin(pi * (10 * a**2 + 37 * a * x + 7 * x**2)))
    print(f)
elif (fun == 'y'):
    y = (log(-5 * a**2 - 16 * a * x + 16 * x**2 + 1) / log(2))
    print(y)
else:
    print("ERROR")

File: clabs/lab1.c

#include <stdio.h>
#include <math.h>

int main(void)
{
	float a, x, s, f, g, y;
	int funct;
	double pi = 3.141592653589;
	printf("Введите x: ");
	scanf("%f", &x);
	printf("Введите a: ");
	scanf("%f", &a);
	printf("Выберите какую ф-ию посчитать: f - 1, g -2, y - 3: ");
	scanf("%i", &funct);

	if (funct == 1) {
		f = (sin(pi * (10 * a * a + 37 * a * x + 7 * x * x)));
		printf("%f\n", f);
	}
	else if (funct == 2) {
		g = -1 * (2 * (-5 * a * a + 3 * a * x + 2 * x * x) / (5 * a * a + 9 * a * x - 2 * x * x));
		printf("%f\n", g);
	}
	else if (funct == 3) {
		y = (log(-5 * a * a - 16 * a * x + 16 * x * x + 1) / log(2));
		printf("%f\n", y);
	}
	else {
		printf("ERROR");
	}
}
